src/shared/seccomp-util.c

```diff
@@ -1061,10 +1061,9 @@ int seccomp_parse_syscall_filter_full(
                                 case -ENOMEM:
                                         return flags & SECCOMP_PARSE_LOG ? log_oom() : -ENOMEM;
                                 case -EEXIST:
-                                        if (flags & SECCOMP_PARSE_LOG)
-                                                log_warning("System call %s already blocked with different errno: %d",
-                                                            name, PTR_TO_INT(hashmap_get(filter, INT_TO_PTR(id + 1))));
-                                        return -EINVAL;
+                                        assert_se(hashmap_update(filter, INT_TO_PTR(id + 1), INT_TO_PTR(errno_num)) == 0);
+                                        r = 0;
```

Comment:
I thought it would be cleaner to ensure a stale `r` “error” isn’t left behind, in case the function’s control flow changes later or something, but I’ll remove it :)